### PR TITLE
MINOR: Use KafkaConsumer in GetOffsetShell

### PR DESCRIPTION
**Pull request by ijuma**
This does the minimal amount of work so that the tool
relies on public non-deprecated APIs (i.e. it no longer
relies on Scala clients code).

Additional improvements (not included here) have
been proposed via KIP-308.

There are a few other PRs that touch this class with
overlapping goals:

- https://github.com/apache/kafka/pull/2891
- https://github.com/apache/kafka/pull/3051
- https://github.com/apache/kafka/pull/3320

One of them remains relevant in the context of KIP-308, but
the others have been superseded. I included the authors of
the 3 PRs as co-authors.

Co-authored-by: Arseniy Tashoyan <tashoyan@gmail.com>
Co-authored-by: Vahid Hashemian <vahidhashemian@us.ibm.com>
Co-authored-by: Mohammed Amine GARMES
Co-authored-by: Ismael Juma <ismael@juma.me.uk>

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 1]
```diff
@@ -69,41 +71,69 @@ object GetOffsetShell {
...
-    val nOffsets = options.valueOf(nOffsetsOpt).intValue
-    val maxWaitMs = options.valueOf(maxWaitMsOpt).intValue()
-
-    val topicsMetadata = ClientUtils.fetchTopicMetadata(Set(topic), metadataTargetBrokers, clientId, maxWaitMs).topicsMetadata
-    if(topicsMetadata.size != 1 || !topicsMetadata.head.topic.equals(topic)) {
-      System.err.println(("Error: no valid topic metadata for topic: %s, " + " probably the topic does not exist, run ").format(topic) +
-        "kafka-list-topic.sh to verify")
-      Exit.exit(1)
+    options.valueOf(nOffsetsOpt).intValue
+    options.valueOf(maxWaitMsOpt).intValue()
```
Comment (vahidhashemian) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala]: Should the above two lines be removed?

Comment (ijuma) [reply core/src/main/scala/kafka/tools/GetOffsetShell.scala]: Yes, will remove.

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 2]
```diff
@@ -47,19 +49,19 @@ object GetOffsetShell {
...
+    parser.accepts("max-wait-ms", "DEPRECATED AND IGNORED: The max amount of time each fetch request waits.")
                            .withRequiredArg
                            .describedAs("ms")
                            .ofType(classOf[java.lang.Integer])
                            .defaultsTo(1000)
-                           
-   if(args.length == 0)
+
+   if (args.length == 0)
       CommandLineUtils.printUsageAndDie(parser, "An interactive shell for getting consumer offsets.")
```
Comment (omkreddy) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala]: nit:  we can update existing usage description 
"An interactive shell for getting TOPIC offsets."

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 3]
```diff
@@ -69,41 +71,67 @@ object GetOffsetShell {
...
-    val time = options.valueOf(timeOpt).longValue
-    val nOffsets = options.valueOf(nOffsetsOpt).intValue
-    val maxWaitMs = options.valueOf(maxWaitMsOpt).intValue()
-
-    val topicsMetadata = ClientUtils.fetchTopicMetadata(Set(topic), metadataTargetBrokers, clientId, maxWaitMs).topicsMetadata
-    if(topicsMetadata.size != 1 || !topicsMetadata.head.topic.equals(topic)) {
-      System.err.println(("Error: no valid topic metadata for topic: %s, " + " probably the topic does not exist, run ").format(topic) +
-        "kafka-list-topic.sh to verify")
-      Exit.exit(1)
+    val partitionIdsRequested = options.valueOf(partitionOpt).split(",").map(_.toInt).toSet
```
Comment (omkreddy) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala]: need to handle empty string.  partitionOpt default value is empty string

File: core/src/main/scala/kafka/tools/GetOffsetShell.scala [hunk 4]
```diff
@@ -69,41 +71,67 @@ object GetOffsetShell {
...
+    val listOffsetsTimestamp = options.valueOf(timeOpt).longValue
+
+    val config = new Properties
+    config.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    config.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, clientId)
```
Comment (omkreddy) [inline core/src/main/scala/kafka/tools/GetOffsetShell.scala:94]: missing consumer deserializer props

**Comment by ijuma** [comment]
Thanks for the review @omkreddy. Sorry, I should have mentioned that this was still WIP and I had not tested it yet. :)

**Review by rajinisivaram — APPROVED** [review]
@ijuma Thanks for the PR, LGTM

**Comment by ijuma** [comment]
JUnit test failure is unrelated and get_offset_shell system test passed:

https://jenkins.confluent.io/view/All/job/system-test-kafka-branch-builder/1788/console

Merging to trunk and 2.0.

**Pull request merged**